chainstate/src/detail/tests/double_spend_tests.rs

```diff
@@ -220,6 +220,97 @@ fn spend_bigger_output_in_the_same_block(#[case] seed: Seed) {
     });
 }
 
+// Try to use the transaction output twice in one block.
+//
+// +--Block----------------+
+// |                       |
+// | +-------tx-1--------+ |
+// | |input = prev_block | |
+// | +-------------------+ |
+// |                       |
+// | +-------tx-2--------+ |
+// | |input = tx1        | |
+// | |input = tx1        | |
+// | +-------------------+ |
+// +-----------------------+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn duplicate_input_in_the_same_tx(#[case] seed: Seed) {
+    common::concurrency::model(move || {
+        let mut tf = TestFramework::default();
+
+        let mut rng = make_seedable_rng(seed);
+        let tx1_output_value = rng.gen_range(100_000..200_000);
+        let first_tx = tx_from_genesis(tf.genesis(), &mut rng, tx1_output_value);
+
+        let input = TxInput::new(first_tx.get_id().into(), 0, InputWitness::NoSignature(None));
+        let second_tx = TransactionBuilder::new()
+            .add_input(input.clone())
+            .add_input(input)
```

Comment:
Maybe also test the spend of the same outpoint twice, i.e. using two different witnesses. Not sure how easy will it be to find an example of that currently, is there a way to tinker with the witness? You can put some data into `InputWitness::NoSIgnature` but not sure what's that for.

Reply:
Optional data is exactly for that purpose